locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+        25  voluntary context switches
+    629566  involuntary context switches
+```
+
+The maximum byte size occupied by virtual memory turns out to be approximately
+`1.52 gb`.
+
+Without streams in place to delegate the backpressure, there is a memory space 
+with an entire degree of magnitude greater that is being allocated. That is a 
+huge margin of difference between the exact process.
+
+This experiment shows how optimized and cost-effective Node's backpressure
+mechanism is for your computing system. Now, let's do a break down on how it 
+works and how we can make sure we are utilizing it in our own applications when 
+building custom streams!
+
+## How Does Backpressure Resolve These Issues?
+
+There are different functions to transfer data from one process to another. In
+Node.js, there is an internal built-in function called [`.pipe()`][]. There are
+[other packages][] out there you can use too! Ultimately though, at the basic
+level of this process, we have two separate components: the _source_ of the
+data and the _consumer_.
+
+When [`.pipe()`][] is called from the source, it signals to the consumer that
+there is data to be transferred. The pipe function helps to set up the
+appropriate backpressure closures for the event triggers.
+
+In Node.js the source is a [`Readable`][] stream and the consumer is the
+[`Writable`][] stream (both of these may be interchanged with a [`Duplex`][] or
+a [`Transform`][] stream, but that is out-of-scope for this guide).
+
+The moment that backpressure is triggered can be narrowed exactly to the return
+value of a `Writable`'s [`.write()`][] function. This return value is
+determined by a few conditions, of course.
+
+In any scenario where the data buffer has exceeded the [`highWaterMark`][] or
+the write queue is currently busy, [`.write()`][] will return `false`.
+
+When a `false` value is returned, the backpressure system kicks in. It will
+pause the incoming [`Readable`][] stream from sending any data and wait until
+the consumer is ready again. Once the data buffer is emptied, a [`.drain()`][] 
+event will be emitted and resume the incoming data flow.
+
+The `Readable` stream also plays a vital role in backpressure. Because the two
+processes rely on one another to communicate effectively, if the `Readable` does
+not respect when the `Writable` stream asks to stop sending data, it can be
+just as problematic.
+
+An instance of this can be unconditionally calling your `Readable` stream to
+send data using the [`push method`][] or forcing data through whenever it
+is available (signaled by the [`.data` event][]):
+
+```javascript
+// This is problematic as it ignores backpressure signals
+readable.on('readable', () => {
+
+  let chunk;
+  while (null !== (chunk = readable.read())) {
+    readable.push(chunk);
+  }
+
+});
+
+// Also ignores backpressure and rams data through, regardless if destination
+// stream is ready or not.
+readable.on('data', data =>
+  writable.write(data);
+);
+```
```

Comment:
Good examples in general, but might be misplaced. This section is about how backpressure resolves these problems, while this portrays counterexamples. Perhaps "Rules to Abide For Writing Custom Streams" will be a better place to put them.